.travis.yml

```diff
@@ -3,6 +3,9 @@ language: cpp
 sudo: required
 dist: trusty
 
+git:
+  depth: 1
```

Comment:
The issue with setting git depth to 1 is if multiple PRs are merged in succession to master, then any queued master builds will fail.  That's why we only enable building last commit on pull requests and not branches.